.github/workflows/build_and_deploy_ext2_img.yml

```diff
@@ -0,0 +1,133 @@
+name: Create and deploy ext2 image to pages
+
+# Controls when the workflow will run
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+  
+  # Lets you supply arguments when you run a github action manually.
+    inputs:
+      DOCKER_BUILD_CONTEXT:
+        type: string
+        description: 'Path or url to the docker build context.'
+        required: true
+        default: 'dockerfiles/example'
+  
+      IMAGE_SIZE:
+        type: string
+        description: 'Size of the base ext2 image. 1GB is the max size for pages, 2GB for Gh release.. Refer to the fallocate(1) manpage for multiplicative suffixes'
+        required: true
+        default: '950MB'
+
+      IMAGE_NAME:
+        type: string
+        description: 'Name of final image. Example final url: https://www.github.io/<owner>/<repositoryName>/<image_name>'
+        required: true
+        default: 'base_image.ext2'
+
+      DEPLOY_TO_PAGES:
+        type: boolean
+        description: 'Deploy to pages.'
+        required: true
+        default: true
+
+      GITHUB_RELEASE:
+        type: boolean
+        description: 'Upload github release.'
+        required: true
+        default: true
+
+      GITHUB_RELEASE_TAG:
+        type: string
+        description: 'Tag for the github release. Optional. Defaults to ext2_image'
+        required: false
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest # Image to run the worker on.
+    
+    # Our variables
+    env:
+      DOCKER_BUILD_CONTEXT: '${{ github.event.inputs.DOCKER_BUILD_CONTEXT }}'
+      TAG: "ext2-webvm-base-image"
+      IMAGE_SIZE: '${{ github.event.inputs.IMAGE_SIZE }}'
+      IMAGE_NAME: '${{ github.event.inputs.IMAGE_NAME }}'
+      DEST: /assets/images/tmp/ # Path to directory where we host the final image from.
+
+    permissions:
+      contents: write    # Required permission to make a github release.  
+  
+    steps:
+      # Checks-out our repository under $GITHUB_WORKSPACE, so our job can access it
+      - uses: actions/checkout@v3
+      # We make the directory to host our image from.
+      - run: sudo mkdir -p $DEST
+      # We build the i386 Dockerfile image.
+      - run: docker build $DOCKER_BUILD_CONTEXT --tag $TAG --platform=i386
+      # We run the image so that we can export the container.
+      - run: docker run -d $TAG
```

Comment:
Something I am missing, do we even need to run the image? Can't we just use `docker save` instead of `docker export` on the image?

Reply:
Docker save and export have different results, as save will save all the docker image its layers and metadata in order for Docker to load it into another docker daemon. Whereas export exclusively exports a container's filesystem its contents. I don't believe it is feasible to use docker save instead for this use-case.

Reply:
Ok, please leave a comment about this